### ISSUE TITLE
Node JS Base64Url encoding is broken

### ISSUE THREAD

**Issue by mikael-lundin** [InteractiveAdvertisingBureau/iabtcf-es#147]
**Version**
TCF 2.0

**Module (core, cmpapi, cli, stub, or testing)**
Core

**Describe with reproduction steps – What is the expected behavior?**
I've managed to encode a string from a TCModel, finally, usinig this code:

```
import { TCString, TCModel, GVL } from '@iabtcf/core';
import vendor from './vendorList.json'
import base64url from 'base64url'
// https://vendorlist.consensu.org/v2/vendor-list.json

const gvl = new GVL(vendor);
const tcModel = new TCModel(gvl);
tcModel.cmpId = 235
tcModel.cmpVersion = 1
const encodedTCString = TCString.encode(tcModel);
const string = base64url.encode(encodedTCString)
console.log(string)

const myTCModel = TCString.decode(string);
console.log(myTCModel)
```

What I get out from that is my  model when it has decoded the string, it looks like this at the start:

```
{
  isServiceSpecific_: false,
  supportOOB_: true,
  useNonStandardStacks_: false,
  purposeOneTreatment_: false,
  publisherCountryCode_: 'AA',
  version_: 2,
  consentScreen_: 0,
  policyVersion_: 2,
  consentLanguage_: 'EN',
  cmpId_: 0,
  cmpVersion_: 0,
  vendorListVersion_: 0,
  specialFeatureOptins: e { bitLength: 0, maxId_: 0, set_: Set {} },
  purposeConsents: e { bitLength: 0, maxId_: 0, set_: Set {} },
  purposeLegitimateInterests: e { bitLength: 0, maxId_: 0, set_: Set {} },
  publisherConsents: e {
    bitLength: 24,

```

I notice that the `cmpId_` is set to 0 which is a bit wierd for me.

also if I do not use `base64url.encode` I get a string that looks like this:

`l.�=-'�i�]%ȫ�`

is that by design or am I missing any step?

**Comment by chrispaterson** [comment]
You're double encoding the `tcModel`.  Your code should be:
```javascript
import { TCString, TCModel, GVL } from '@iabtcf/core';
import vendor from './vendorList.json'

const gvl = new GVL(); // I removed vendor, I'm not sure what that was for
const tcModel = new TCModel(gvl);
tcModel.cmpId = 235
tcModel.cmpVersion = 1
const encodedTCString = TCString.encode(tcModel);
console.log(encodedTCString)

const myTCModel = TCString.decode(encodedTCString);
console.log(myTCModel)
```

**Comment by mikael-lundin** [comment]
hi, using your code I get this error:
```
Error [GVLError]: must specify GVL.baseUrl before loading GVL json
```
It states in the documentation that you must provide a vendorlist JSON when creating a GVL.

**Comment by chrispaterson** [comment]
I see it now.  Let me investigate this...

**Comment by mikael-lundin** [comment]
Also I see in your encoder you use the function `btoa` it's not really native to node, so the recommendation would be to use `Buffer.from(char).toString('base64')`

**Comment by chrispaterson** [comment]
@mikael-lundin I re-wrote the Base64Url encoder class to not use either `atob`, `btoa`, or `Buffer`.  I had previously written it this way and then was convinced to use the standard implementation from the browser because I thought it would be less maintenance.  It turns out that is complicates things much more and bloats the code a lot.  Please check out this pull request and see if this implementation works for your environment:
https://github.com/InteractiveAdvertisingBureau/iabtcf-es/pull/152

**Comment by mikael-lundin** [comment]
Shit yeah! it works :D I got this now:
```
COx8C9bOx8C9bDrACAENAgCAAAAAAAAAAAAAF3EXySoGY2tho2YVFzBEIYwfJxyigMgShgQIsS0NQIeFLBoGPiAAHBGYJAQAGBAkkACBAQIsHGBMCQABgAgRiRCMQEGMDzNIBIBAggkbI0FACCVmnkHS3ZCY70-6u__QAAAAAAAA.IF3EXySoGY2tho2YVFzBEIYwfJxyigMgShgQIsS0NQIeFLBoGPiAAHBGYJAQAGBAkkACBAQIsHGBMCQABgAgRiRCMQEGMDzNIBIBAggkbI0FACCVmnkHS3ZCY70-6u__QAAA
```
BTW, the build showed something like 80 errors but it still managed to build it. :)

**Comment by chrispaterson** [comment]
@mikael-lundin It's released as version 1.0.1-0
